packages/resolve-scripts/src/server/auth/index.js

```diff
@@ -1,6 +1,53 @@
-import { getRouteByName } from './helper'
-import localStrategy from './localStrategy'
-import githubStrategy from './githubStrategy'
-import googleStrategy from './googleStrategy'
+import localStrategyCallbackCreator from './strategy-callback-creators/local'
+import azureadStrategyCallbackCreator from './strategy-callback-creators/azuread'
+import oauthStrategyCallbackCreator from './strategy-callback-creators/oauth'
 
-export { getRouteByName, localStrategy, githubStrategy, googleStrategy }
+const strategyCallbackCreators = {
+  local: localStrategyCallbackCreator,
+  'azuread-openidconnect': azureadStrategyCallbackCreator,
+  github: oauthStrategyCallbackCreator,
+  google: oauthStrategyCallbackCreator
+}
+
+const getStarategyName = (PassportStrategy, options) => {
+  const strategy = new PassportStrategy(
```

Comment:
It is expensive method to find name, isn't it?
Besides that it is hidden dependency from passport's inner algo.

Reply:
Yes, because there isn't any cheap method